cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorTest.java

```diff
@@ -231,6 +231,58 @@ public void testTimeoutLeaderActions() throws InterruptedException {
     waitUntilExecutionFinishes(executor);
   }
 
+  @Test
+  public void testForceStopExecution() throws InterruptedException {
+    createTopics();
+    // The proposal tries to swap replicas.
+    ExecutionProposal proposal =
+        new ExecutionProposal(TP1, 0, new ReplicaPlacementInfo(1),
+            Arrays.asList(new ReplicaPlacementInfo(0), new ReplicaPlacementInfo(1)),
+            Arrays.asList(new ReplicaPlacementInfo(1), new ReplicaPlacementInfo(0)));
+
+    KafkaCruiseControlConfig configs = new KafkaCruiseControlConfig(getExecutorProperties());
+    Time time = new MockTime();
+    MetadataClient mockMetadataClient = EasyMock.createMock(MetadataClient.class);
+    // Fake the metadata to never change so the leader movement will timeout.
+    Node node0 = new Node(0, "host0", 100);
+    Node node1 = new Node(1, "host1", 100);
+    Node[] replicas = new Node[2];
+    replicas[0] = node0;
+    replicas[1] = node1;
+    PartitionInfo partitionInfo = new PartitionInfo(TP1.topic(), TP1.partition(), node1, replicas, replicas);
+    Cluster cluster = new Cluster("id", Arrays.asList(node0, node1), Collections.singleton(partitionInfo),
+                                  Collections.emptySet(), Collections.emptySet());
+    MetadataClient.ClusterAndGeneration clusterAndGeneration = new MetadataClient.ClusterAndGeneration(cluster, 0);
+    EasyMock.expect(mockMetadataClient.refreshMetadata()).andReturn(clusterAndGeneration).anyTimes();
+    EasyMock.expect(mockMetadataClient.cluster()).andReturn(clusterAndGeneration.cluster()).anyTimes();
+    EasyMock.replay(mockMetadataClient);
+
+    Collection<ExecutionProposal> proposalsToExecute = Collections.singletonList(proposal);
+    Executor executor = new Executor(configs, time, new MetricRegistry(), mockMetadataClient, 86400000L,
+                                     43200000L, null, getMockUserTaskManager(RANDOM_UUID),
```

Comment:
Nit: Can we avoid hardcoding constants? -- i.e. can move them to static variables?